processing/src/main/java/io/druid/query/aggregation/SerializablePairLongStringSerde.java

```diff
@@ -0,0 +1,178 @@
+import io.druid.segment.data.GenericIndexed;
+import io.druid.segment.data.ObjectStrategy;
+import io.druid.segment.serde.ComplexColumnPartSupplier;
+import io.druid.segment.serde.ComplexMetricExtractor;
+import io.druid.segment.serde.ComplexMetricSerde;
+import io.druid.segment.serde.LargeColumnSupportedComplexColumnSerializer;
+import io.druid.segment.writeout.SegmentWriteOutMedium;
+
+import java.nio.ByteBuffer;
+
+/**
+ * The SerializablePairLongStringSerde serializes a Long-String pair (SerializablePairLongString).
+ * The serialization structure is: Long:Integer:String
+ * <p>
+ * The class is used on first/last String aggregators to store the time and the first/last string.
+ * Long:Integer:String -> Timestamp:StringSize:StringData
+ */
+public class SerializablePairLongStringSerde extends ComplexMetricSerde
+{
+
+  public static final String TYPE_NAME = "serializablePairLongString";
+
+  @Override
+  public String getTypeName()
+  {
+    return TYPE_NAME;
+  }
+
+  @Override
+  public ComplexMetricExtractor getExtractor()
+  {
+    return new ComplexMetricExtractor()
+    {
+      @Override
+      public Class<SerializablePairLongString> extractedClass()
+      {
+        return SerializablePairLongString.class;
+      }
+
+      @Override
+      public Object extractValue(InputRow inputRow, String metricName)
+      {
+        return inputRow.getRaw(metricName);
+      }
+    };
+  }
+
+  @Override
+  public void deserializeColumn(ByteBuffer buffer, ColumnBuilder columnBuilder)
+  {
+    final GenericIndexed column = GenericIndexed.read(buffer, getObjectStrategy(), columnBuilder.getFileMapper());
+    columnBuilder.setComplexColumn(new ComplexColumnPartSupplier(getTypeName(), column));
+  }
+
+  @Override
+  public ObjectStrategy getObjectStrategy()
+  {
+    return new ObjectStrategy<SerializablePairLongString>()
+    {
+      @Override
+      public int compare(SerializablePairLongString o1, SerializablePairLongString o2)
+      {
+        int comparation;
+
+        // First we check if the objects are null
+        if (o1 == null && o2 == null) {
+          comparation = 0;
+        } else if (o1 == null) {
+          comparation = -1;
+        } else if (o2 == null) {
+          comparation = 1;
+        } else {
+
+          // If the objects are not null, we will try to compare using timestamp
+          comparation = o1.lhs.compareTo(o2.lhs);
+
+          // If both timestamp are the same, we try to compare the Strings
+          if (comparation == 0) {
+
+            // First we check if the strings are null
+            if (o1.rhs == null && o2.rhs == null) {
+              comparation = 0;
+            } else if (o1.rhs == null) {
+              comparation = -1;
+            } else if (o2.rhs == null) {
+              comparation = 1;
+            } else {
+
+              // If the strings are not null, we will compare them
+              // Note: This comparation maybe doesn't make sense to first/last aggregators
```

Comment:
Would you elaborate more on this comment?

Reply:
Yeah sure! 

```
If the strings are not null, we will compare them
Note: This comparison maybe doesn't make sense to first/last aggregators,
because compare both SerializablePairLongString based on String maybe doesn't matter
to define each is first or last.
```